CorsixTH/Lua/HeatingController.lua

```diff
@@ -0,0 +1,182 @@
+--[[ Copyright (c) 2009 Peter "Corsix" Cawley
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+class "HeatingController"
+
+---@type HeatingController
+local HeatingController = _G["HeatingController"]
+
+function HeatingController:HeatingController(world)
+  self.world = world
+  self.radiator_heat = 0.5
+  self:resetBoilerBreakdownCounter()
+  self.acc_heating = 0
+end
+
+function HeatingController:resetBoilerBreakdownCounter()
+  self.days_until_boiler_breakdown = self.world.map.level_config.gbv.DisasterLaunch
+  --if disasterLaunch not defined in Levelfile
+  if self.days_until_boiler_breakdown == nil then
+    self.days_until_boiler_breakdown = 200
+  end
+end
+
+function HeatingController:checkHeatingFacilities(day,averagePatiensWarmth, averageStaffWarmth)
+  if not self.warmth_msg and not self.heating_broke then
+    if day == 15 then
+      if averagePatiensWarmth < 0.22 then
```

Comment:
Can you move these magic numbers to 'constants, e.g. `local COLD_TOLERANCE_TEMP = 0.22`